aws/resource_aws_iam_user.go

```diff
@@ -242,14 +275,29 @@ func resourceAwsIamUserDelete(d *schema.ResourceData, meta interface{}) error {
 		}
 	}
 
-	request := &iam.DeleteUserInput{
+	deleteUserPermissionsBoundaryInput := &iam.DeleteUserPermissionsBoundaryInput{
 		UserName: aws.String(d.Id()),
 	}
 
-	log.Println("[DEBUG] Delete IAM User request:", request)
-	if _, err := iamconn.DeleteUser(request); err != nil {
+	log.Println("[DEBUG] Delete IAM User Permissions Boundary request:", deleteUserPermissionsBoundaryInput)
+	_, err = iamconn.DeleteUserPermissionsBoundary(deleteUserPermissionsBoundaryInput)
```

Comment:
This seems odd that we'd have to delete the boundary before we delete the resource. I'm guessing that deleting a user before the boundary will cause issues but it seems like you'd want two resources here instead of one.